### PR TITLE
added a simple spring boot java app example

### PR DESCRIPTION
**Pull request by johnscancella**
A very simple hello world using java and spring boot

### CODE REVIEW

**Comment by afeld** [comment]
This seems like a lot for a Hello World app...was a lot of it auto-generated or something?

@jcscottiii Being our resident Java expert, mind taking a look?

**Comment by jmcarp** [comment]
There are a few spring examples in https://github.com/cloudfoundry-samples. Speaking of which, could we contribute our examples to `cloudfoundry-samples` instead of managing our own project?

**Comment by afeld** [comment]
Nice! Had no idea that existed. https://github.com/18F/cf-hello-worlds/issues/22

**Comment by LinuxBozo** [comment]
@afeld, couple of files were definitely autogenerated by `gradle`, the build runner. The core of the app itself is only 2 files, `Main.java` and `HelloWorldController.java`, and the test `HelloWorldControllerTest.java`

**Comment by afeld** [comment]
@johnscancella Thoughts about your implementation, vs. something like https://github.com/cloudfoundry-samples/hello-spring? Maybe we transfer this pull request there instead...?

**Comment by LinuxBozo** [comment]
@afeld The difference between this example, and the [example at cloudfoundry-samples](https://github.com/cloudfoundry-samples/hello-spring) could set off a holy war.. Maven vs Gradle. ;-) 

Also, this example is Spring Boot, the other is Spring MVC. Both Spring, but slightly different. Boot is current gen tech, and can configure/integrate MVC.

**Comment by DavidEBest** [comment]
Given that some of the files are generated, is it necessary to include the `gradlew` and `gradlew.bat` files?

**Comment by jcscottiii** [comment]
@DavidEBest yeah you need those two. essentially, if you don't have gradle installed locally yourself, you can just run the one for your OS and it'll be fine

File: java/src/test/java/gov/eighteenf/controller/HelloWorldControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  private MockMvc mvc;
+
+  @Before
+  public void setUp() throws Exception {
+    mvc = MockMvcBuilders.standaloneSetup(new HelloWorldController()).build();
+  }
+
+  @Test
+  public void getHello() throws Exception {
+    mvc.perform(MockMvcRequestBuilders.get("/").accept(MediaType.APPLICATION_JSON))
```
Comment (jcscottiii) [inline java/src/test/java/gov/eighteenf/controller/HelloWorldControllerTest.java]: two options:
1. you can remove the `.accept(MediaType.APPLICATION_JSON)` part
2. or you also check the ContentType `.andExpect(content().contentType()`

File: java/README.MD [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#Deploying the hello world Java App
+
+###Build the application
+```bash
+run ./gradle distZip
```
Comment (jcscottiii) [inline java/README.MD]: there's no `gradle` file but you can do 
```bash
# windows
$  ./gradlew.bat distZip
# Mac OS X / Linux 
$  ./gradlew distZip
```

**Comment by jcscottiii** [comment]
@johnscancella this looks good. :) you need to also include the files from the `gradle/wrapper` folder

**Comment by johnscancella** [comment]
@afeld Yeah it is a bit verbose, but I didn't want anyone to have to install gradle. If you(or @jmcarp, @LinuxBozo, @jcscottiii) think that is an ok requirement to install gradle I can remove them.
@jcscottiii thanks for the compliments, updated as you suggested.

**Comment by johnscancella** [comment]
Ohh and @LinuxBozo yes definitely holy war when it comes to maven vs. gradle. I much prefer gradle, but if you think it will make a difference I can try and generate a maven pom file.

**Comment by afeld** [comment]
This works—we can always iterate, and maybe transfer this to a `spring-boot-gradle` repository under @cloudfoundry-samples.

**Comment by afeld** [comment]
Thanks @johnscancella for the contribution, and thanks everyone for the input!

**Comment by johnscancella** [comment]
no problem, glad I could help. +1 on merging with the cloudfoundry-samples at some point

**Pull request merged**